### PR TITLE
Feature/peer

### PR DESCRIPTION
**Pull request by paulfariello**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+language: python
+python:
+  - "3.6"
+install:
+  - pip install nose twine
```
Comment (SabatierBoris) [inline .travis.yml]: Why nose ?
Why not use develop_dependencie in the setup.py ?
If you still want to use nose, you can say it in the setup.py so you can run `./setup.py test`

File: stormbot/storage.py [hunk 2]
```diff
@@ -68,5 +71,17 @@ def proxy(self, value):
         return value
 
     def dump(self):
-        with open(self.path, 'w') as cachefile:
-            json.dump(self._cache, cachefile, cls=ProxyEncoder)
+        self._file.seek(0)
+        json.dump(self._cache, self._file, cls=ProxyEncoder)
+
+def mock(values):
```
Comment (SabatierBoris) [inline stormbot/storage.py]: Why this code isn't in the test directory ?

File: stormbot/bot.py [hunk 3]
```diff
@@ -162,19 +275,156 @@ def subscribe(self, nick, plugin):
...
+            return self._peers.values()
+        else:
+            distribution = pkg_resources.get_distribution(plugin.__class__.__module__)
+            return filter(lambda p: p.supports(distribution.project_name, distribution.version), self._peers.values())
+
 class Fakebot:
     def write(sef, *args, **kwargs):
         print(*args, **kwargs)
 
+def mock(argv):
```
Comment (SabatierBoris) [inline stormbot/bot.py]: Why this code isn't in the test directory ?

File: .travis.yml (hunk 1 above)
Comment (paulfariello) [reply .travis.yml]: I could remove nose and just use unittest.

**Pull request merged**